### PR TITLE
feat: Add InfluxDb module

### LINKED ISSUE #950: [Enhancement]: Add InfluxDB module
### Problem

There is no support for the InfluxDB container.

### Solution

Introduce the InfluxDB module.

### Benefit

We will be able to run integration tests for services that use InfluxDB.

### Alternatives

No alternatives.

### Would you like to help contributing this enhancement?

Yes

### PR DESCRIPTION
**Pull request by MelomanG**
## What does this PR do?
This PR adds support for starting [InfluxDb] containers.

## Why is it important?
This is required to write tests that target a InfluxDb database.

## Related issues
- Closes #950

## How to test this PR
cd tests/Testcontainers.InfluxDb.Tests
dotnet test

### CODE REVIEW

File: src/Testcontainers.InfluxDb/InfluxDbBuilder.cs [hunk 1]
```diff
@@ -0,0 +1,168 @@
...
+        return base.Init()
+            .WithImage(InfluxDbImage)
+            .WithPortBinding(InfluxDbPort, true)
+            .WithEnvironment("DOCKER_INFLUXDB_INIT_MODE", "setup")
+            .WithUsername(DefaultUsername)
+            .WithPassword(DefaultPassword)
+            .WithOrganization(DefaultOrganization)
+            .WithBucket(DefaultBucket)
+            .WithWaitStrategy(Wait.ForUnixContainer().UntilHttpRequestIsSucceeded(request =>
+                request.ForPort(InfluxDbPort).ForPath("/ping").ForStatusCode(HttpStatusCode.NoContent)));
```
Comment (HofmeisterAn) [inline src/Testcontainers.InfluxDb/InfluxDbBuilder.cs:126]: Doesn't Java set up basic HTTP authentication using the username and password configuration? Is it not necessary?

File: tests/Testcontainers.InfluxDb.Tests/InfluxDbContainerTests.cs [hunk 2]
```diff
@@ -0,0 +1,31 @@
...
+    {
+        // Given
+        using var client = new InfluxDBClient(_influxDbContainer.GetAddress());
+
+        // When
+        var result = await client.PingAsync()
+            .ConfigureAwait(false);
+
+        // Then
+        Assert.True(result);
```
Comment (HofmeisterAn) [inline tests/Testcontainers.InfluxDb.Tests/InfluxDbContainerTests.cs]: Can we add a slightly more complex test that requires authentication (the tests are usually a good start or example for other developers )?

**Review by HofmeisterAn — COMMENTED** [review]
Thank you for the pull request. The majority of it looks good. I've removed the BOMs and applied some common code styles. I've also included two minor suggestions. This is a great first Testcontainers for .NET contribution 👍. Thanks again.

File: src/Testcontainers.InfluxDb/InfluxDbBuilder.cs (hunk 1 above)
Comment (MelomanG) [reply src/Testcontainers.InfluxDb/InfluxDbBuilder.cs:126]: For `/ping` endpoint as far as I understand it is not required to provide any token.
https://docs.influxdata.com/influxdb/v2.7/api/#operation/GetPing

File: tests/Testcontainers.InfluxDb.Tests/InfluxDbContainerTests.cs (hunk 2 above)
Comment (MelomanG) [reply tests/Testcontainers.InfluxDb.Tests/InfluxDbContainerTests.cs]: Done ;)

File: src/Testcontainers.InfluxDb/InfluxDbBuilder.cs (hunk 1 above)
Comment (HofmeisterAn) [reply src/Testcontainers.InfluxDb/InfluxDbBuilder.cs:126]: > The influx ping command checks the health of a running InfluxDB instance by querying the /health endpoint. It does not require an API token.

You are right, thanks for double-checking.

File: tests/Testcontainers.InfluxDb.Tests/InfluxDbContainerTests.cs (hunk 2 above)
Comment (HofmeisterAn) [reply tests/Testcontainers.InfluxDb.Tests/InfluxDbContainerTests.cs]: Thanks

**Review by HofmeisterAn — APPROVED** [review]
🎉 thanks!

**Pull request merged**